web/src/mobile/components/WalletDepositBody/index.tsx

```diff
@@ -1,74 +1,99 @@
-        });
+    useEffect(() => {
+        setTab(currencyItem?.networks ? currencyItem?.networks[0]?.blockchain_key?.toUpperCase() : '');
+    }, [wallet.currency]);
 
-        const buttonLabel = `${intl.formatMessage({ id: 'page.body.wallets.tabs.deposit.ccy.button.generate' })} ${wallet.currency.toUpperCase()} ${intl.formatMessage({ id: 'page.body.wallets.tabs.deposit.ccy.button.address' })}`;
+    const depositAddress = wallet.deposit_addresses?.find(address => address.blockchain_key?.toLowerCase() === tab?.toLowerCase());
 
-        if (wallet.type === 'coin') {
-            return (
-                <React.Fragment>
-                    <CurrencyInfo wallet={wallet}/>
-                    {currencyItem && !currencyItem.deposit_enabled ? (
-                        <Blur
-                            className={blurCryptoClassName}
-                            text={intl.formatMessage({ id: 'page.body.wallets.tabs.deposit.disabled.message' })}
-                        />
-                    ) : null}
+    const text = intl.formatMessage({ id: 'page.body.wallets.tabs.deposit.ccy.message.submit' },
+        { confirmations: currencyItem.min_confirmations });
+    const error = intl.formatMessage({id: 'page.body.wallets.tabs.deposit.ccy.message.pending'});
+
+    const title = intl.formatMessage({ id: 'page.body.wallets.tabs.deposit.fiat.message1' });
+    const description = intl.formatMessage({ id: 'page.body.wallets.tabs.deposit.fiat.message2' });
+
+    const buttonLabel = `${intl.formatMessage({ id: 'page.body.wallets.tabs.deposit.ccy.button.generate' })} ${wallet.currency.toUpperCase()} ${intl.formatMessage({ id: 'page.body.wallets.tabs.deposit.ccy.button.address' })}`;
+
+    const handleGenerateAddress = useEffect(() => {
+        if (!depositAddress && wallet.type !== 'fiat') {
+            dispatch(walletsAddressFetch({ currency: wallet.currency, blockchain_key: tab }));
+        }
+    }, [wallet, walletsAddressFetch, tab]);
+
+    const onTabChange = label => {
+        const blockchain = currencyItem.networks?.find(item => item.protocol.toUpperCase() === label);
+
+        setTab(blockchain.blockchain_key);
+    };
+
+    const onCurrentTabChange = index => setCurrentTabIndex(index);
+
+    const renderTabs = useMemo(() => {
+        return currencyItem.networks?.map(network => {
+            return {
+                content: tab.toUpperCase() === network.blockchain_key?.toUpperCase() ?
                     <DepositCrypto
                         buttonLabel={buttonLabel}
                         copiableTextFieldText={`${wallet.currency.toUpperCase()} ${label}`}
                         copyButtonText={intl.formatMessage({ id: 'page.body.wallets.tabs.deposit.ccy.message.button'} )}
                         error={error}
-                        handleGenerateAddress={props.handleGenerateAddress}
+                        handleGenerateAddress={() => handleGenerateAddress}
```

Comment:
maybe you mean
```suggestion
                        handleGenerateAddress={() => handleGenerateAddress()}
```
but anyway, what is the purpose of generating arrow func here?